pkg/telemetry/platform.go

```diff
@@ -0,0 +1,99 @@
+package telemetry
```

Comment:
why not have it in https://github.com/nginxinc/telemetry-exporter ?

Reply:
My initial thought is since the code has to deal with checking for the K8s platform (with the purpose of collecting telemetry data) it doesn't seem the best fit in a project which focuses on telemetry exporting. I feel as if it seems like it might just be putting it there for the purpose of putting shared code in a separate shared project. If there was a shared collection project that would be great, but there isn't. Do you think that thought is not applicable and that it would be better to put it in the other project?

Reply:
@pleshakov responded offline saying that
> the exporter also defines common data points. So having shared code that helps to collect those data points, from that point of view, it makes sense to me

@shaun-nx do you have any thoughts on this?

Reply:
Decided offline to move code to `internal` directory as this code will not be shared with NIC.